ui/app/components/mfa-setup-step-one.js

```diff
@@ -0,0 +1,75 @@
+import Component from '@glimmer/component';
+import { inject as service } from '@ember/service';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+
+/**
+ * @module MfaSetupStepOne
+ * MfaSetupStepOne component is a child component used in the end user setup for MFA. It records the UUID (aka method_id) and sends a admin-generate request.
+ *
+ * @param {string} entityId - the entityId of the user. This comes from the auth service which records it on loading of the cluster. A root user does not have an entityId.
+ * @param {function} isUUIDVerified - a function that consumes a boolean. Is true if the admin-generate is successful and false if it throws a warning or error.
+ * @param {boolean} restartFlow - a boolean that is true that is true if the user should proceed to step two or false if they should stay on step one.
+ * @param {function} saveUUIDandQrCode - A function that sends the inputted UUID and return qrCode from step one to the parent.
+ * @param {boolean} showWarning - whether a warning is returned from the admin-generate query. Needs to be passed to step two.
+ */
+
+export default class MfaSetupStepOne extends Component {
+  @service store;
+  @tracked error;
+  @tracked warning = '';
+  @tracked qrCode = '';
+
+  @action
+  redirectPreviousPage() {
+    this.args.restartFlow();
+    window.history.back();
```

Comment:
later we can work on a service that records route history and allows us to go to the previous route. For now, given our time constraints this will work.